tmt/steps/prepare/artifact/providers/repository_provider.py

```diff
@@ -1,108 +1,139 @@
     def _download_artifact(
-        self, artifact: RepositoryFileInfo, guest: Guest, destination: Path
+        self, artifact: RpmArtifactInfo, guest: Guest, destination: Path
     ) -> None:
-        url = artifact.location
-        filename = artifact.id
-        repo_destination = Path("/etc/yum.repos.d") / filename
-        sudo_prefix = "sudo " if not guest.facts.is_superuser else ""
-
-        self.logger.debug(f"Processing repository file from {url} to {repo_destination} on guest.")
+        """This provider only sets up the repo, it does not download RPMs."""
+        self.logger.debug(f"Skipping download of '{artifact.id}'.")
+
+    def download_artifacts(
+        self,
+        guest: Guest,
+        download_path: Path,
+        exclude_patterns: list[Pattern[str]],
+    ) -> list[Path]:
+        """
+        Download the .repo file to the guest, making the repository available.
+
+        This method overrides the default behavior to prevent downloading
+        individual RPMs. It installs the repository, lists the available
+        packages for discovery, and then returns.
+        """
+        if self._repository_installed:
+            self.logger.debug("Repository file already installed, skipping.")
+            return []
+
+        # 1. Install the repository file on the guest using info from our helper object
+        filename = self.repo_file.filename
+        url = self.repo_file.url
+        repo_dest = Path("/etc/yum.repos.d") / filename
+
+        sudo = "sudo " if not guest.facts.is_superuser else ""
+        self.logger.info(f"Installing repository '{url}' to '{repo_dest}'.")
 
         try:
-            # Download the file to the temporary destination
+            # TODO: Add retry Mechanism
             guest.execute(
-                ShellScript(f"curl -L --fail -o {quote(str(destination))} {quote(url)}"),
+                ShellScript(f"{sudo}curl -L --fail -o {quote(str(repo_dest))} {quote(url)}"),
                 silent=True,
             )
+            self._repository_installed = True
+        except GeneralError as error:
+            raise DownloadError(f"Failed to download repository file to '{repo_dest}'.") from error
 
-            # Move the file to /etc/yum.repos.d and set permissions
-            guest.execute(
-                ShellScript(
-                    f"""
-                 set -e
-                 {sudo_prefix}mv {quote(str(destination))} {quote(str(repo_destination))}
-                 {sudo_prefix}chmod 644 {quote(str(repo_destination))}
-                """
-                ),
-                silent=True,
-            )
+        # 2. Populate the RPM list for discovery purposes
+        self._fetch_rpms(guest, repo_dest)
 
-        except GeneralError as e:
-            raise DownloadError(
-                f"Failed to process repository file from {url} to {repo_destination}."
-            ) from e
+        self.logger.info("Repository setup is complete.")
+        return [repo_dest]
```

Comment:
The `return` will have to be revised. It currently has to be the list of rpm files, not the repo destination. But that is likely to change when we rename the functions.